lib/kubernetes-deploy/kubectl.rb

```diff
@@ -24,40 +27,37 @@ def run(*args, log_failure: nil, use_context: true, use_namespace: true, output:
       raise_if_not_found: false, attempts: 1, output_is_sensitive: nil)
```

Comment:
I refactored this method a little to make it easier to follow (and accommodate adding a retry on timeout). There are two small functional changes that I think are good things:
- Previously, if log_failure && not_404 && debug_logging_enabled, the error would be printed twice, once at the warn level and once at the debug level. Now it will only be printed at the warn level.
- Previously, if log_failure && 404 && raise_on_404, the failed command would be logged. Now it won't be.